### DISCUSSION TITLE
Compiled arm64 Mac binary is actually x86

### DISCUSSION THREAD

**Discussion by drewcassidy** [pypa/cibuildwheel#1071] [Q&A] [answered]
This may be an issue with my build setup, but since it compiles fine on my own M1 machine I'm putting it here:

The "repair" step of CIBW fails when building on arm64 macOS, saying that the repair step failed to find the arm64 binary

see: https://github.com/drewcassidy/quicktex/runs/5797453843?check_suite_focus=true or
[logs_7.zip](https://github.com/pypa/cibuildwheel/files/8401833/logs_7.zip) if it expires

I don't think I'm doing anything unusual, but since nobody else seems to have this issue I don't know if its a bug. My workflow is here: https://github.com/drewcassidy/quicktex/blob/e488dbcbffe088d87c10db9fb0003f83bc30a02e/.github/workflows/python-package.yml#L47-L99

Any help?

**Comment by henryiii** [comment, marked as answer]
If you manually implement CMake building instead of using scikit-build, you'll need to handle the M1 support manually. Compare https://github.com/drewcassidy/quicktex/blob/e488dbcbffe088d87c10db9fb0003f83bc30a02e/setup.py with https://github.com/pybind/cmake_example/blob/544111c7d6184559b54c5d8c6b1a613314904b0a/setup.py#L94-L98 (other parts too, but IIRC those are the important lines).

**Reply by drewcassidy** [reply]
Thanks for the quick response! I'll try that out tomorrow and report back

**Reply by drewcassidy** [reply]
That worked! thank you!
